src/layout/Table/TableComponent.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+
+import { Delete as DeleteIcon } from '@navikt/ds-icons';
+
+import { AppTable } from 'src/app-components/table/Table';
+import { Caption } from 'src/components/form/caption/Caption';
+import { FD } from 'src/features/formData/FormDataWrite';
+import { useDataModelBindings } from 'src/features/formData/useDataModelBindings';
+import { Lang } from 'src/features/language/Lang';
+import { useLanguage } from 'src/features/language/useLanguage';
+import { useIsMobile } from 'src/hooks/useDeviceWidths';
+import { useNodeItem } from 'src/utils/layout/useNodeItem';
+import type { TableActionButton } from 'src/app-components/table/Table';
+import type { PropsFromGenericComponent } from 'src/layout';
+import type { IDataModelReference } from 'src/layout/common.generated';
+import type { LayoutNode } from 'src/utils/layout/LayoutNode';
+
+type TableComponentProps = PropsFromGenericComponent<'Table'>;
+
+type TableSummaryProps = {
+  componentNode: LayoutNode<'Table'>;
+};
+
+export function TableSummary({ componentNode }: TableSummaryProps) {
+  const item = useNodeItem(componentNode);
+  const { formData } = useDataModelBindings(item.dataModelBindings, 1, 'raw');
+  const { title } = item.textResourceBindings ?? {};
+  const { langAsString } = useLanguage();
+  const isMobile = useIsMobile();
+
+  const data = formData.simpleBinding;
+
+  if (!Array.isArray(data)) {
+    return null;
+  }
+
+  if (data.length < 1) {
+    return null;
+  }
+  return (
+    <AppTable<IDataModelReference>
```

Comment:
This is surely not displaying a table for a `IDataModelReference`? That's config, and not meant to be exposed to the user in any way. Since you `useDataModelBindings(..., 'raw')` above, the data is `FDValue` (or whatever that type is called), which is practically almost the same as `any`.

Another way of putting it; maybe this doesn't need to use generics at all, or at least provide a default value when the generic value is not used for type-safety at all (like is the case here).

Reply:
Yeah I think a key valye type of some sort is better :)

Reply:
Yep - but make sure the component also handles nested arrays, objects, null, etc without failing. :raised_hands: